### ISSUE TITLE
permission error for 9p sync folder on linux mint host

### ISSUE THREAD

**Issue by brainchild0** [vagrant-libvirt/vagrant-libvirt#1430]
Attempting to provision a synced folder by 9p, I find the following error message when creating the machine:

```
/home/user/.vagrant.d/gems/2.7.4/gems/fog-libvirt-0.9.0/lib/fog/libvirt/requests/compute/vm_action.rb:7:in `create': Call to virDomainCreateWithFlags failed: internal error: qemu unexpectedly closed the monitor: 2022-01-11T13:13:41.684974Z qemu-system-x86_64: -device virtio-9p-pci,id=fs0,fsdev=fsdev-fs0,mount_tag=7570cc6d848c322dd078d6e22d7437c,bus=pci.0,addr=0x5: cannot initialize fsdev 'fsdev-fs0': failed to open '/path/to/share': Permission denied (Libvirt::Error)
```

The message is proceeded by a long stack trace from Ruby.

Following is the configuration line from the `Vagrant` file:

```
config.vm.synced_folder "/path/to/share", "/srv",
                        type: "9p", accessmode: "mapped", mount: false

```

The host directory is owned by the user invoking Vagrant, and the objective is to expose it on the guest as being owned by user `vagrant`, with new files created on either side reflecting the same mapping.

I tested both with `mount` enabled and disabled, with no noticeable difference. Frankly, I am confused about the meaning of this value.

The guest is built from `debian/bullseye64`. The host is Linux Mint 20.3. All relevant packages on the host are installed from the distribution, except Vagrant itself, which comes from the vendor's Ubuntu repository, and the plugin, which was added using the `plugin` command executed by the user, because the `vagrant-libvirt` package from the vendor is not causing the plugin to be detected by Vagrant.

**Comment by electrofelix** [comment]
I've just tested the following Vagrantfile locally:
```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.memory = 2048
    libvirt.cpus = 2
    libvirt.uri = 'qemu:///session'
  end

  config.vm.box = "generic/debian11"
  config.vm.synced_folder "./", "/srv", type: "9p", accessmode: "mapped", mount: true
end
```

I don't appear to get the stacktrace that you did, but it also doesn't map as the vagrant user. Reading the 9p docs isn't suggesting to me that it would either. Instead files created under the /srv as root in the guest appear on the host with the same uid as my host user.

Not sure that 9p is going to necessarily work here.

**Comment by electrofelix** [comment]
Forgot to add `mount` controls whether the filesystem will be mounted automatically in the machine or not.

One possible solution here is to use bindfs to mount a fuse driver over it to translate to the needed uid within the machine.
e.g.
```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.memory = 2048
    libvirt.cpus = 2
  end

  config.vm.box = "generic/debian11"
  config.vm.synced_folder "./", "/mnt/.srv", type: "9p", accessmode: "mapped", mount: true

  config.vm.provision "shell", privileged: false, inline: <<-EOF
  sudo apt-get update
  sudo apt-get install -y bindfs
  sudo bindfs -u $(id -u) -g $(id -g) /mnt/.srv/ /srv/
  EOF
end
```

End up using an intermediary folder of `/mnt/.srv` and then bindfs exposes this with the vagrant uid/gid under /srv. Note the use of `privileged: false` to allow the `$(id -u):$(id -g)` be resolved by the shell for the default user before being passed via sudo to bindfs. If you want to run it privileged you'll need to use `$(id -u vagrant):$(id -g vagrant)` instead.

**Comment by electrofelix** [comment]
So my assumption here for the stack trace is that it was likely due to the use of user session qemu and referencing a folder on the host to which the user does not have sufficient permissions to access, leading to libvirt/qemu having an error when attempting to mount.

I can emulate this by using a directory such as `/var/lib/polkit-1/` where:
```
ls -la /var/lib/polkit-1/
ls: cannot open directory '/var/lib/polkit-1/': Permission denied
```
within a vagrantfile
```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.memory = 2048
    libvirt.cpus = 2
    libvirt.qemu_use_session = true
  end

  config.vm.box = "generic/debian11"
  config.vm.synced_folder "/var/lib/polkit-1", "/srv", type: "9p", accessmode: "mapped"

end
```

And I will get the following traceback.
```
Bringing machine 'default' up with 'libvirt' provider...
==> default: Checking if box 'generic/debian11' version '4.1.16' is up to date...
==> default: Creating image (snapshot of base box volume).
==> default: Creating domain with the following settings...
==> default:  -- Name:              vagrant-libvirt_default
==> default:  -- Description:       Source: /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/Vagrantfile
==> default:  -- Domain type:       kvm
==> default:  -- Cpus:              2
==> default:  -- Feature:           acpi
==> default:  -- Feature:           apic
==> default:  -- Feature:           pae
==> default:  -- Clock offset:      utc
==> default:  -- Memory:            2048M
==> default:  -- Management MAC:    
==> default:  -- Loader:            
==> default:  -- Nvram:             
==> default:  -- Base box:          generic/debian11
==> default:  -- Storage pool:      default
==> default:  -- Image(vda):        /home/user/.local/share/libvirt/images/vagrant-libvirt_default.img, virtio, 128G
==> default:  -- Disk driver opts:  cache='default'
==> default:  -- Kernel:            
==> default:  -- Initrd:            
==> default:  -- Graphics Type:     vnc
==> default:  -- Video Type:        cirrus
==> default:  -- Video VRAM:        256
==> default:  -- Video 3D accel:    false
==> default:  -- Sound Type:        
==> default:  -- Keymap:            en-us
==> default:  -- TPM Backend:       passthrough
==> default:  -- TPM Path:          
==> default:  -- INPUT:             type=mouse, bus=ps2
==> default: Removing 9p devices for shared folders...
==> default: Configuring 9p devices for shared folders...
==> default: Creating shared folders metadata...
==> default: Starting domain.
==> default: Removing domain...
==> default: Deleting the machine folder
Traceback (most recent call last):
	82: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/batch_action.rb:86:in `block (2 levels) in run'
	81: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/machine.rb:203:in `action'
	80: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/machine.rb:203:in `call'
	79: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/environment.rb:631:in `lock'
	78: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/machine.rb:217:in `block in action'
	77: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/machine.rb:248:in `action_raw'
	76: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `run'
	75: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/util/busy.rb:19:in `busy'
	74: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `block in run'
	73: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builder.rb:180:in `call'
	72: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	71: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	70: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	69: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/box_check_outdated.rb:93:in `call'
	68: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	67: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/call.rb:53:in `call'
	66: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `run'
	65: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/util/busy.rb:19:in `busy'
	64: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `block in run'
	63: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builder.rb:180:in `call'
	62: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	61: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	60: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	59: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/cleanup_on_failure.rb:21:in `call'
	58: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	57: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/provision.rb:80:in `call'
	56: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	55: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/set_name_of_domain.rb:34:in `call'
	54: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	53: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/handle_storage_pool.rb:63:in `call'
	52: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	51: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
	50: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	49: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/handle_box_image.rb:127:in `call'
	48: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	47: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/create_domain_volume.rb:94:in `call'
	46: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	45: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/resolve_disk_settings.rb:141:in `call'
	44: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	43: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/create_domain.rb:442:in `call'
	42: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	41: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/create_networks.rb:40:in `call'
	40: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	39: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/create_network_interfaces.rb:190:in `call'
	38: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	37: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	36: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	35: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/call.rb:53:in `call'
	34: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `run'
	33: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/util/busy.rb:19:in `busy'
	32: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `block in run'
	31: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builder.rb:180:in `call'
	30: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	29: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	28: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	27: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/call.rb:53:in `call'
	26: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `run'
	25: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/util/busy.rb:19:in `busy'
	24: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/runner.rb:101:in `block in run'
	23: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builder.rb:180:in `call'
	22: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	21: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	20: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	19: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/prepare_nfs_valid_ids.rb:14:in `call'
	18: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	17: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/plugins/synced_folders/nfs/action_cleanup.rb:25:in `call'
	16: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	15: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/synced_folder_cleanup.rb:28:in `call'
	14: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	13: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/delayed.rb:19:in `call'
	12: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	11: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/builtin/synced_folders.rb:87:in `call'
	10: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	 9: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/prepare_nfs_settings.rb:21:in `call'
	 8: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	 7: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/share_folders.rb:22:in `call'
	 6: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	 5: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/set_boot_order.rb:22:in `call'
	 4: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/bundler/gems/vagrant-c9ce3319d42c/lib/vagrant/action/warden.rb:48:in `call'
	 3: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/lib/vagrant-libvirt/action/start_domain.rb:476:in `call'
	 2: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/gems/fog-libvirt-0.9.0/lib/fog/libvirt/models/compute/server.rb:76:in `start'
	 1: from /home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/gems/fog-libvirt-0.9.0/lib/fog/libvirt/requests/compute/vm_action.rb:7:in `vm_action'
/home/user/git/github.com/vagrant-libvirt/vagrant-libvirt/vendor/bundle/ruby/2.6.0/gems/fog-libvirt-0.9.0/lib/fog/libvirt/requests/compute/vm_action.rb:7:in `create': Call to virDomainCreateWithFlags failed: internal error: qemu unexpectedly closed the monitor: 2022-10-13T12:06:58.615194Z qemu-system-x86_64: -device virtio-9p-pci,id=fs0,fsdev=fsdev-fs0,mount_tag=ba5c9a3a0741eb0ce0f36b5f05b0574,bus=pci.0,addr=0x6: cannot initialize fsdev 'fsdev-fs0': failed to open '/var/lib/polkit-1': Permission denied (Libvirt::Error)
```

This is because when you use sessions the qemu process executes with the same permissions as your current user, while when you use `uri = qemu:///system` or `qemu_use_session = false`, it will execute with root permissions.

I think the only fix needed here is possibly a validation check for the synced folders in the case of when `qemu_use_session = true` to check if the user is allowed access the host path before attempting to bring up.

Otherwise for permissions in the guest it's the same issue as #378
